Compiler/runtime/systemimpl.c

```diff
@@ -2230,6 +2230,7 @@ extern char* SystemImpl__iconv(const char * str, const char *from, const char *t
   iconv_t ic;
   int count;
   char *buf;
+  if (!strcmp(from, to)) return (char*)str;
```

Comment:
@sjoelund: yes, i got it. but can be moved further below, or we could check if the output is the same and free the memory.